### PR TITLE
403 only get needed mailboxes polling

### PR DESCRIPTION
**Pull request by alagane**

### CODE REVIEW

**Comment by chibenwa** [comment]
I do not succeed to know which commit to review.

Can you rebase (reword?) the commits parts of other PRs, not specific to this work to add a `[NO_REVIEW]` prefix so that reviewers do not end up wasting there time reviewing the same commits several time?

**Comment by alagane** [comment]
Force-pushed for the rebase ;)
Every commit is to review now, though if you want to see the most important part you might check 4ca5097fac25ff4284efdce1348077d543dfe521

**Comment by chibenwa** [comment]
Reviewed.

**Comment by alagane** [comment]
It seems there is a problem introduced by this PR:
When loading http://localhost:9900/inbox/#/unifiedinbox/inbox URL, the assignMailbox function will be called without context thus the request will not work.
I don't know why it was not called before the request and why it is called now.
There is no problem when context is given (clicking a mailbox on the list, or even all mailboxes works), I am investigating.

**Comment by chibenwa** [comment]
@tuanlc would this qualify for a review?

**Comment by alagane** [comment]
> It seems there is a problem introduced by this PR:
> When loading http://localhost:9900/inbox/#/unifiedinbox/inbox URL, the assignMailbox function will be called without context thus the request will not work.
> I don't know why it was not called before the request and why it is called now.
> There is no problem when context is given (clicking a mailbox on the list, or even all mailboxes works), I am investigating.

Ok, it seems this is not related, and introduced by 1a93a6c60e1d3585a4afd72f66b270ae90828058

**Comment by alagane** [comment]
There are still issues when removing a folder, the folder is still here (but put at the end of the list?)
I do not know why, it seems it is not on the cache any more, neither on the mailboxes on the view scope, and I don't know where it comes from.

**Comment by alagane** [comment]
> There are still issues when removing a folder, the folder is still here (but put at the end of the list?)
> I do not know why, it seems it is not on the cache any more, neither on the mailboxes on the view scope, and I don't know where it comes from.

Might be related to this already existing bug #413

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js [hunk 1]
```diff
@@ -253,62 +260,85 @@ angular.module('esn.inbox.libs')
...
+
+        if (_.isEmpty(mailboxesToFetch)) {
+          // in this case the event is not broadcasted by _updateMailboxCache but needed
+          $rootScope.$broadcast(INBOX_EVENTS.FOLDERS_UPDATED);
+        }
+      }
+
+      if (!_.isEmpty(mailboxesToFetch)) {
+        return jmapClient
+          .mailbox_get({
```
Comment (fabienmoyon) [inline src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:306]: you don't need to go to a new line here

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js [hunk 2]
```diff
@@ -253,62 +260,85 @@ angular.module('esn.inbox.libs')
...
+            accountId: null,
+            ids: mailboxesToFetch
+          })
+          .then(function(mailboxes) {
+            const translatedMailboxList = _translateMailboxes(mailboxes.list);
+
+            return _addSharedMailboxVisibility(translatedMailboxList);
+          })
+          .then(function(mailboxListWithSharedVisibility) {
+
```
Comment (fabienmoyon) [inline src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:316]: you don't need a empty line here

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js [hunk 3]
```diff
@@ -253,62 +260,85 @@ angular.module('esn.inbox.libs')
...
+    }
+
+    function _updateCacheFetchingAllMailboxes(jmapClient) {
 
-      _updateMailboxCache(sharedMailboxList);
+      return jmapClient.mailbox_get({
+        accountId: null,
+        ids: null
+      }).then(function(mailboxes) {
+
```
Comment (fabienmoyon) [inline src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:328]: here also

**Review by fabienmoyon — DISMISSED** [review]

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js (hunk 1 above)
Comment (alagane) [reply src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js]: Ok, I don't know about this one, but I fixed it.

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js (hunk 2 above)
Comment (alagane) [reply src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js]: Fixed

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js (hunk 3 above)
Comment (alagane) [reply src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js]: Fixed

**Review by oubchid — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**